core/src/read/read_stream.rs

```diff
@@ -756,58 +756,84 @@ impl<D: Decoder> ReadDiskStream<D> {
             reached_end_of_file = true;
         }
 
+        let copy_block_into_read_buffer =
```

Comment:
Why not extract this function at the module level instead of using a closure?

Reply:
Since it is only used by this one method, I think it makes sense to use a closure. But if you think it would be more readable as a function, we could do that.

Reply:
Ok, I extracted it into a function and did that let-else thing. You're right, I think it is easier to read that way.